web-common/src/features/canvas/stores/canvas-stores.ts

```diff
@@ -0,0 +1,47 @@
+// CanvasStore.ts
+import { derived, writable, type Readable } from "svelte/store";
+import { CanvasEntity } from "./canvas-entity";
+
+export interface CanvasStoreType {
+  entities: Record<string, CanvasEntity>;
+}
+
+function createCanvasStore() {
+  const { subscribe, update } = writable<CanvasStoreType>({
+    entities: {},
+  });
+
+  // Add a new CanvasEntity to the store by name
+  function addEntity(name: string) {
```

Comment:
Nit: "addCanvas" and "removeCanvas" would be more semantic. I'd expect "addEntity" only if the entity type were variable.